File: puzzle1/puzzle1.ex
defmodule Puzzle1 do
    def calculate_fuel_sum do
        {:ok, contents} = File.read("puzzle1_input.txt")
        masses = contents
            |> String.split
            |> Enum.map(&String.to_integer/1)
        Enum.reduce(masses, 0, &((div(&1, 3) - 2) + &2))
    end
end

IO.puts Puzzle1.calculate_fuel_sum

File: puzzle1/puzzle_1.hs
#!/usr/bin/env stack
import System.IO

main :: IO ()
main = do
    puzzleInput <- readFile "puzzle1_input.txt"
    putStrLn $ show $ calculateFuelSum $ map (read::String->Int) $ lines puzzleInput

calculateFuelSum :: [Int] -> Int
calculateFuelSum xs = foldr (\x y -> (x `div` 3 - 2) + y) 0 xs

